src/test/fixtures/utils.js

```diff
@@ -18,33 +18,43 @@ export function getBoundingBox(width: number, height: number) {
   };
 }
 
+/**
+ * This function formats a call tree into a human readable form, to make it easy
+ * to assert certain relationships about the data structure in a really terse
+ * and human-friendly fashion. For instance a call tree could become formatted
+ * like so:
+ *
+ * [
+ *   '- A (total: 4, self:—)',
+ *   '  - B (total: 3, self:—)',
+ *   '    - C (total: 1, self:1)',
+ *   '    - D (total: 1, self:1)',
+ *   '    - E (total: 1, self:1)',
+ *   '  - F (total: 1, self:1)',
+ * ]
+ *
+ * This structure is easy to read, avoids whitespace issues, and diffs really well
+ * on the test output, showing where errors occur. Previously snapshots were used,
+ * but the assertion was hidden in another file, which really hurt discoverability
+ * and maintainability.
+ */
 export function formatTree(
   callTree: CallTree,
   children: IndexIntoCallNodeTable[] = callTree.getRoots(),
   depth: number = 0,
-  previousString: string = ''
-) {
-  const whitespace = Array(depth * 2).join(' ');
+  lines: string[] = []
+): string[] {
+  const whitespace = Array(depth * 2 + 1).join(' ');
```

Comment:
look, starting from node v8, we can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart :)

What do you think about upgrading node to latest LTS (8.9) in circle.yml and appveyor.yml and then use `padStart` here?

Reply:
I'd prefer not to take on this work in this PR.

Reply:
I meant: maybe upgrade node in a separate PR before landing this one.
But I don't really mind for this, I'm just a bit sad ;)